### ISSUE TITLE
Decrementing the invoice counter by -1 not working

### ISSUE THREAD

**Issue by BeckeBauer** [kimai/kimai#3885]
### Describe the issue

- access demo version [https://demo-plugins.kimai.org/](https://demo-plugins.kimai.org/) 
- create 2 invoices (2023/001 and 2023/002)
- klick ![image](https://user-images.githubusercontent.com/51003649/222109118-e83b25a4-c035-48da-8a7f-52ad3947c56a.png) and change the invoice number format to `{Y}/{cy-1,3}` 
- create third invoice for different customer
expected behaviour: second invoice 2023/002
factual behaviour: invoice 2023/003
(since I accidentally assigned the same invoice number twice and manually added the second invoice to the database, I need to decrease the counter by 1 for the consecutive invoices)

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

1.30.10

### How do you run Kimai?

Other (please mention below)

### Which PHP version are you using?

8.1

**Comment by kevinpapst** [comment]
Kimai will not create duplicate invoice numbers! 

> create third invoice for different customer

That's irrelevant, unless you use a different counter type like {ccy}.

> expected behaviour: second invoice 2023/002

Nope. I mean, maybe for you. But not from a technical perspective. Duplicate invoice numbers are not allowed:

Kimai sees 3 invoices. 
Decrements by 1 = 2. 
Tests if that invoice number exists => yes.
Tries the next number => 3, still exists. 
Tries the next number => 2. 
Does not exist: uses 4 to create the invoice.

**Comment by BeckeBauer** [comment]
In the documents, it is said:
```
Assume that you want to change your invoice numbering and reset it to zero, you already wrote 72 invoices this year, and your counter is an incrementing number per year that should be prefixed with the four digit year:

{Y}/{cy-72,3} would result in 2020/001 for your first invoice
```
So, that only works if the 72 invoices were created by other tools than Kimai? With other words: decrementing works only if invoices were not created with Kimai?

**Comment by kevinpapst** [comment]
No, it always works, but Kimai will deny to create duplicate invoice numbers.
If the first 72 were created with `{cy}` then the next ones can use `{Y}/{cy-72,3}` and so start at `2023-001` instead of `2023-073`.

In other words: Incrementing and especially decrementing works only in certain scenarios.

**Comment by BeckeBauer** [comment]
I am still confused: 
In my case, the invoice number format is `{Y}-{cy,3}-{ccy,2}` (which "translates" to 2023-001-01 in case of the first invoice in 2023). 

With Kimai, I created the invoice 2023-120-02 for _customer A_ and outside Kimai the invoice 2023-120-03 for _customer B_ (accidently, both invoices use `120` but are otherwise different). 

I added manually the invoice `2023-120-03` for _customer B_ to the Kimai database. 

Now, I want Kimai to use 2023-**121**-XX for the next invoice and thought to achieve it be adding -1: `{Y}-{cy-1,3}-{ccy,2}`

However, the next invoice uses 2023-**122**-XX although there is no 2023-121-XX (at least not as visible invoice number).

Should I give up or is there something I can do to "convince" Kimai to assign 121 to the next invoice?
